packages/shared/src/components/post/PostActions.tsx

```diff
@@ -42,95 +37,29 @@ export function PostActions({
   onBookmark,
   origin = Origin.ArticlePage,
 }: PostActionsProps): ReactElement {
-  const { trackEvent } = useContext(AnalyticsContext);
-  const { user, showLogin } = useContext(AuthContext);
-  const { updatePost } = useUpdatePost();
   const { data, onShowPanel, onClose } = useBlockPostPanel(post);
   const { showTagsPanel } = data;
-  const onUpvotePostMutate = updatePost({
-    id: post.id,
-    update: mutationHandlers.upvote(post),
-  });
-  const onDownvoteMutate = updatePost({
-    id: post.id,
-    update: mutationHandlers.downvote(post),
-  });
-  const onCancelDownvoteMutate = updatePost({
-    id: post.id,
-    update: mutationHandlers.cancelDownvote(post),
-  });
-  const { upvotePost, cancelPostUpvote, downvotePost, cancelPostDownvote } =
-    useVotePost({
-      onUpvotePostMutate: (params) => {
-        onClose(true);
-        return onUpvotePostMutate(params);
-      },
-      onCancelPostUpvoteMutate: updatePost({
-        id: post.id,
-        update: mutationHandlers.cancelUpvote(post),
-      }),
-      onDownvotePostMutate: (params) => {
-        onShowPanel();
-        return onDownvoteMutate(params);
-      },
-      onCancelPostDownvoteMutate: (params) => {
-        onClose(true);
-        return onCancelDownvoteMutate(params);
-      },
-    });
 
-  const toggleUpvote = () => {
-    if (user) {
-      if (post?.userState?.vote === UserPostVote.Up) {
-        trackEvent(
-          postAnalyticsEvent(AnalyticsEvent.RemovePostUpvote, post, {
-            extra: { origin },
-          }),
-        );
-        return cancelPostUpvote({ id: post.id });
-      }
-      if (post) {
-        trackEvent(
-          postAnalyticsEvent(AnalyticsEvent.UpvotePost, post, {
-            extra: { origin },
-          }),
-        );
-        return upvotePost({ id: post.id });
-      }
-    } else {
-      showLogin(AuthTriggers.Upvote);
-    }
-    return undefined;
-  };
+  const { toggleUpvote, toggleDownvote } = useVotePost();
 
-  const toggleDownvote = () => {
-    if (!post) {
-      return;
+  const onToggleUpvote = async () => {
+    if (post?.userState?.vote === UserPostVote.None) {
+      onClose(true);
     }
 
-    if (!user) {
-      showLogin(AuthTriggers.Downvote);
-
-      return;
-    }
+    await toggleUpvote({ post, origin });
+  };
 
+  const onToggleDownvote = async () => {
     if (post?.userState?.vote === UserPostVote.Down) {
-      trackEvent(
-        postAnalyticsEvent(AnalyticsEvent.RemovePostDownvote, post, {
-          extra: { origin },
-        }),
-      );
-
-      cancelPostDownvote({ id: post.id });
-    } else {
-      trackEvent(
-        postAnalyticsEvent(AnalyticsEvent.DownvotePost, post, {
-          extra: { origin },
-        }),
-      );
+      onClose(true);
+    }
 
-      downvotePost({ id: post.id });
+    if (post?.userState?.vote === UserPostVote.None) {
+      onShowPanel();
     }
```

Comment:
Is this the new experiment panel?
If so it should also be under a flag.

Reply:
No, this is the block report panel after downvote.